test/e2e/specs/editor/various/post-visibility.spec.js

```diff
@@ -0,0 +1,125 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Post visibility', () => {
+	const wp = '';
+	test.afterEach( async ( { pageUtils } ) => {
+		await pageUtils.setBrowserViewport( 'large' );
+	} );
+	[ 'large', 'small' ].forEach( ( viewport ) => {
+		test( `can be changed when the viewport is ${ viewport }`, async ( {
+			page,
+			admin,
+			pageUtils,
+			editor,
+		} ) => {
+			await pageUtils.setBrowserViewport( viewport );
+
+			await admin.createNewPost();
+
+			await editor.openDocumentSettingsSidebar();
+
+			await page.click(
+				'.components-button.edit-post-post-visibility__toggle.is-tertiary'
+			);
+
+			await page.click( '#editor-post-private-0' );
+
+			await page.click( 'role=button[name="OK"i]' );
+
+			const currentStatus = await page.evaluate( () => {
+				return wp.data
+					.select( 'core/editor' )
+					.getEditedPostAttribute( 'status' );
+			} );
+
+			expect( currentStatus ).toBe( 'private' );
+		} );
+
+		test( `can be canceled when the viewport is ${ viewport }`, async ( {
+			page,
+			pageUtils,
+			admin,
+			editor,
+		} ) => {
+			await pageUtils.setBrowserViewport( viewport );
+
+			await admin.createNewPost();
+
+			await editor.openDocumentSettingsSidebar();
+
+			const initialStatus = await page.evaluate( () => {
+				return wp.data
+					.select( 'core/editor' )
+					.getEditedPostAttribute( 'status' );
+			} );
+
+			await page.click(
+				'.components-button.edit-post-post-visibility__toggle.is-tertiary'
+			);
+
+			await page.click( '#editor-post-private-0' );
+
+			await page.click( 'role=button[name="Cancel"i]' );
+
+			const currentStatus = await page.evaluate( () => {
+				return wp.data
+					.select( 'core/editor' )
+					.getEditedPostAttribute( 'status' );
+			} );
+
+			expect( initialStatus ).toBe( currentStatus );
+		} );
+	} );
+
+	test( 'visibility remains private even if the publish date is in the future', async ( {
+		page,
+		admin,
+		editor,
+	} ) => {
+		await admin.createNewPost();
+
+		// Enter a title for this post.
+		await page.type( '.editor-post-title__input', 'Title' );
+
+		await editor.openDocumentSettingsSidebar();
+
+		// Set a publish date for the next month.
+		await page.click(
+			'.components-button.edit-post-post-schedule__toggle.is-tertiary'
+		);
+
+		await page.click(
+			'role=button[name="Move forward to switch to the next month."i]'
```

Comment:
I think the name should be `View next month`. Not sure why it's showing differently on your machine? Maybe a rebase would help?

Reply:
@kevin940726, Rebased the code but still showing the same name. 

![image](https://user-images.githubusercontent.com/41000648/195529738-a336010c-bac8-427c-b9b0-d292ad9a7e7c.png)

Reply:
@kevin940726,

Updated the test case as per the feedbacks. Please check..